executor.py

```diff
@@ -41,23 +41,36 @@ def index(
 
     @requests(on='/search')
     def search(
-        self,
-        docs: Optional['DocumentArray'] = None,
-        parameters: Optional[Dict] = None,
-        **kwargs,
+            self,
+            docs: Optional['DocumentArray'] = None,
+            parameters: Optional[Dict] = None,
+            **kwargs,
     ):
         """Perform a vector similarity search and retrieve the full Document match
 
         :param docs: the Documents to search with
-        :param parameters: the runtime arguments to `DocumentArray`'s match function. They overwrite the original match_args arguments.
+        :param parameters: the runtime arguments to `DocumentArray`'s match
+        function. They overwrite the original match_args arguments.
         """
         if not docs:
             return
         match_args = deepcopy(self._match_args)
         if parameters:
             match_args.update(parameters)
+
+        match_args = SimpleIndexer._filter_parameters(docs, match_args)
+
         docs.match(self._storage, **match_args)
 
+    @staticmethod
+    def _filter_parameters(docs, match_args):
+        # get only those arguments that exist in .match
+        args = set(docs.match.__code__.co_varnames)
+        # and remove self
+        args.discard('self')
```

Comment:
`co_varnames` returns the names of the **local variables** in the function, rather than argument names only. 

```suggestion
        import inspect
        args = inspect.signature(docs.match).parameters
```